src/Compiler/TypedTree/TcGlobals.fs

```diff
@@ -175,6 +175,13 @@ let tname_IsByRefLikeAttribute = "System.Runtime.CompilerServices.IsByRefLikeAtt
 // Table of all these "globals"
 //-------------------------------------------------------------------------
 
+[<RequireQualifiedAccess>]
+type CompilationMode =
+    | Unset
+    | OneOff
+    | Service
```

Comment:
Reminder: The Service and FSI values need to ensure their value is set, I think this is not yet being done.

Reply:
It isn't and I've done it consciously, since we specifically don't use `Service` or `Interactive` cases anywhere, hence `Unset` represents the case of "like it was before"